dotCMS/src/main/java/com/dotmarketing/db/listeners/ReindexListener.java

```diff
@@ -0,0 +1,89 @@
+package com.dotmarketing.db.listeners;
+
+import java.util.List;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.google.common.collect.ImmutableList;
+
+public class ReindexListener implements DotListener {
+
+	public enum Action{ADDING, REMOVING};
+
+	
+	private final Action action;
+	private final List<Contentlet> contentToIndex;
+
+
+
+	public List<Contentlet> getReindexIds() {
+		return contentToIndex;
+	}
+
+    public ReindexListener(final List<Contentlet> reindexIds) {
+        this(reindexIds,Action.ADDING);
+    }
+	
+	public ReindexListener(final List<Contentlet> reindexIds, final Action action) {
+		super();
+		this.contentToIndex = reindexIds;
+		this.action = action;
+	}
+
+	
+    public ReindexListener(Contentlet reindexId) {
+        this(reindexId,Action.ADDING);
+    }
+	
+	public ReindexListener(Contentlet reindexId, Action action) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'action' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)